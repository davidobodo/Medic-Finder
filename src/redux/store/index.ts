import {createStore, applyMiddleware} from "redux";
import createSagaMiddleware from "redux-saga";

import rootReducer from "../reducers";
import rootSaga from "../sagas";

const sagaMiddleware = createSagaMiddleware();

//add saga middleware for taking care of making network requests
const store = createStore(rootReducer, applyMiddleware(sagaMiddleware));

sagaMiddleware.run(rootSaga);

export default store;
