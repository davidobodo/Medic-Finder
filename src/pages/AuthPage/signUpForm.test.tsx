import React from 'react';
import { Provider } from 'react-redux';
import store from '../../redux/store';

import { cleanup, fireEvent, render } from '@testing-library/react';
import '@testing-library/jest-dom';

import SignUp from './signUp/signUp';
import Login from './login/login';

afterEach(cleanup);

//--------------------------------------------------------------
// signup form tests
//--------------------------------------------------------------

const signUpProps = {
	onRequireLogin: jest.fn()
};

const app = (
	<Provider store={store}>
		<SignUp {...signUpProps} />
	</Provider>
);

it('should take a snapshot', () => {
	const { container } = render(app);
	expect(container).toMatchSnapshot();
});

it('First name should change input value', () => {
	const input = render(app).queryByPlaceholderText('First Name') as HTMLInputElement;
	fireEvent.change(input, { target: { value: 'rukky' } });
	expect(input.value).toBe('rukky');
});
