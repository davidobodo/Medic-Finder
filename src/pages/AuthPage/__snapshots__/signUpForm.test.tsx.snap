// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<div>
  <section
    class="makeStyles-formContainer-10"
  >
    <h4
      class="MuiTypography-root makeStyles-header-11 MuiTypography-h4"
    >
      Sign up
    </h4>
    <p
      class="makeStyles-formDescription-12"
    >
      Create your 100% free account and start searching for health facilities near you.
    </p>
    <form
      novalidate=""
    >
      <div
        class="makeStyles-inputContainer-13"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          data-testid="firstName"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
            data-shrink="true"
            for="standard-full-width"
            id="standard-full-width-label"
          >
            First Name
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              id="standard-full-width"
              name="firstName"
              placeholder="First Name"
              type="text"
              value=""
            />
          </div>
          
        </div>
      </div>
      <div
        class="makeStyles-inputContainer-13"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          data-testid="lastName"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
            data-shrink="true"
            for="standard-full-width"
            id="standard-full-width-label"
          >
            Last Name
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              id="standard-full-width"
              name="lastName"
              placeholder="Last Name"
              type="text"
              value=""
            />
          </div>
          
        </div>
      </div>
      <div
        class="makeStyles-inputContainer-13"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          data-testid="email"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
            data-shrink="true"
            for="standard-full-width"
            id="standard-full-width-label"
          >
            Email
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              id="standard-full-width"
              name="email"
              placeholder="email"
              type="email"
              value=""
            />
          </div>
          
        </div>
      </div>
      <div
        class="makeStyles-inputContainer-13"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          data-testid="password"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
            data-shrink="true"
            for="standard-full-width"
            id="standard-full-width-label"
          >
            Password
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              id="standard-full-width"
              name="password"
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
          
        </div>
      </div>
      <div
        class="makeStyles-inputContainer-13"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          data-testid="confirmPassword"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
            data-shrink="true"
            for="standard-full-width"
            id="standard-full-width-label"
          >
            Confirm Password
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              id="standard-full-width"
              name="confirmPassword"
              placeholder="Confirm Password"
              type="password"
              value=""
            />
          </div>
          
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained Mui-disabled Mui-disabled"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Sign Up
        </span>
      </button>
    </form>
    <div
      class="makeStyles-enquiry-14"
    >
      Already have an account? 
      <button>
        Login
      </button>
    </div>
  </section>
</div>
`;
